core/base/planarGraphLayout/MergeTreeVisualization.h

```diff
@@ -0,0 +1,930 @@
+    ImportantPairsSpacing = d;
+  }
+  void setNonImportantPairsSpacing(double d) {
+    NonImportantPairsSpacing = d;
+  }
+  void setNonImportantPairsProximity(double d) {
+    NonImportantPairsProximity = d;
+  }
+
+  // ==========================================================================
+  // Planar Layout
+  // ==========================================================================
+  // TODO manage multi pers pairs
+  template <class dataType>
+  void treePlanarLayoutBDImpl(
+    FTMTree_MT *tree,
+    std::vector<float> &retVec,
+    std::vector<LongSimplexId> &treeSimplexId,
+    std::vector<idNode> &branching,
+    std::vector<std::vector<ftm::idNode>> &nodeBranching) {
+    idNode treeRoot = tree->getRoot();
+    idNode treeRootOrigin = tree->getNode(treeRoot)->getOrigin();
+    float rootY = retVec[treeSimplexId[treeRoot] * 2 + 1];
+    float rootOriginY = retVec[treeSimplexId[treeRootOrigin] * 2 + 1];
+    float rootYmin = std::min(rootY, rootOriginY);
+    float rootYmax = std::max(rootY, rootOriginY);
+    dataType rootPers = tree->getNodePersistence<dataType>(treeRoot);
+    std::vector<std::tuple<float, float>> allNodeSpanX(
+      tree->getNumberOfNodes());
+    std::vector<std::tuple<float, float>> allNodeImportantSpanX(
+      tree->getNumberOfNodes());
+
+    // Compute gap
+    float nonImportantPairsGap
+      = (rootYmax - rootYmin) * 0.05 * NonImportantPairsSpacing;
+    float importantPairsGap
+      = std::max(nonImportantPairsGap, (float)ImportantPairsSpacing);
+
+    // Some functions
+    auto comp = [&](const ftm::idNode a, const ftm::idNode b) {
+      return tree->getNodePersistence<dataType>(a)
+             < tree->getNodePersistence<dataType>(b);
+    };
+    auto compSup = [&](const ftm::idNode a, const ftm::idNode b) {
+      return not comp(a, b);
+    };
+
+    // Go
+    std::vector<ftm::idNode> leaves = tree->getLeavesFromTree();
+    sort(leaves.begin(), leaves.end(), comp);
+    std::queue<ftm::idNode> queue;
+    for(auto node : leaves)
+      queue.push(node);
+    while(!queue.empty()) {
+      idNode node = queue.front();
+      queue.pop();
+      idNode nodeOrigin = tree->getNode(node)->getOrigin();
+
+      dataType nodePers = tree->getNodePersistence<dataType>(node);
+      retVec[treeSimplexId[nodeOrigin] * 2] = 0;
+      retVec[treeSimplexId[nodeOrigin] * 2 + 1]
+        = nodePers / rootPers * (rootYmax - rootYmin) + rootYmin;
+
+      // Positioning nodes in the branch
+      if(tree->isBranchOrigin(nodeOrigin)) {
+        float prevX = 0;
+        std::vector<idNode> nodeBranchingVector;
+        for(unsigned int i = 1; i < nodeBranching[nodeOrigin].size(); ++i)
+          nodeBranchingVector.push_back(nodeBranching[nodeOrigin][i]);
+        sort(nodeBranchingVector.begin(), nodeBranchingVector.end(), compSup);
+
+        // Iterate through each node of the branch
+        int lastIndexImportant = -1;
+        for(unsigned int i = 0; i < nodeBranchingVector.size(); ++i) {
```

Comment:
prefer size_t for size related variables

Reply:
Done in the next commit!